osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs

```diff
@@ -28,8 +28,10 @@ public TaikoPerformanceCalculator(Ruleset ruleset, WorkingBeatmap beatmap, Score
         {
         }
 
-        public override double Calculate(Dictionary<string, double> categoryDifficulty = null)
+        public override double Calculate(Dictionary<string, double> categoryDifficulty = null, double upTo = Double.PositiveInfinity)
         {
+            Attributes = (TaikoDifficultyAttributes)Ruleset.CreateDifficultyCalculator(Working).Calculate(upTo, Score.Mods);
```

Comment:
I've had this throw an invalid cast exception when [this branch](https://github.com/Poyo-SSB/osu/blob/partial-difficulty-performance-calculation/osu.Game.Rulesets.Taiko/Difficulty/TaikoDifficultyCalculator.cs#L49) in TaikoDifficultyCalculator was hit, which returns DifficultyAttributes instead of TaikoDifficultyAttributes. May be worth looking in to.

Reply:
See PR #4220 for fix.